apps/aecore/src/aec_state.erl

```diff
@@ -123,10 +120,11 @@ code_change(_OldVsn, State, _Extra) ->
 setup_trees(0, Trees) ->
     Trees;
 setup_trees(N, Trees) ->
+    TreesAtEndOfPreviousBlock = setup_trees(N-1, Trees),
     {ok, Block} = aec_chain:get_block_by_height(N),
     Txs = aec_blocks:txs(Block),
-    {ok, TreesUpdated} = apply_txs_internal(Txs, N, Trees),
-    setup_trees(N-1, TreesUpdated).
```

Comment:
OK, something here is not correct, but we need recursion to initialize state from block at given height.

This should go from genesis to N, not from N to genesis.

Reply:
Recursion is in line 123.

Reply:
It is using stack.

Reply:
ok, sorry I missed that. I will refactor to stack-less solution. It may be very heavy with long chaing

Reply:
ok

Reply:
@sennui Feel free to just close this PR and include/squash some/all these changes in PR 147 (even force-pushing is ok).

Reply:
merging this, fixing and will squash